pkg/controller/windowsmachineconfig/nodeconfig/node_config.go

```diff
@@ -0,0 +1,139 @@
+package nodeconfig
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	certificates "k8s.io/api/certificates/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// RetryCount is the amount of times we will retry an api operation
+	RetryCount = 20
+	// RetryInterval is the interval of time until we retry after a failure
+	RetryInterval = 5 * time.Second
+)
+
+// nodeConfig holds the information to make the given VM a kubernetes node. As of now, it holds the information
+// related to kubeclient,
+type nodeConfig struct {
+	k8sclientset *kubernetes.Clientset
+}
+
+func NewNodeConfig(clientset *kubernetes.Clientset) *nodeConfig {
+	return &nodeConfig{k8sclientset: clientset}
+}
+
+var log = logf.Log.WithName("nodeconfig")
+
+// HandleCSRs handles the approval of bootstrap and node CSRs
+func (nc *nodeConfig) HandleCSRs() error {
+	// Handle the bootstrap CSR
+	err := nc.handleCSR("system:serviceaccount:openshift-machine-config-operator:node-bootstrapper")
+	if err != nil {
+		return fmt.Errorf("unable to handle bootstrap CSR: %v", err)
+	}
+
+	// Handle the node CSR
+	// Note: for the product we want to get the node name from the instance information
+	err = nc.handleCSR("system:node:")
+	if err != nil {
+		return fmt.Errorf("unable to handle node CSR: %v", err)
+	}
+	return nil
+}
+
+// findCSR finds the CSR that matches the requestor filter
```

Comment:
This finds the CSR that contains the requestor filter, not matches. Seems like this could cause issues if we're not looking for an exact match